libminifi/test/script-tests/ExecutePythonProcessorTests.cpp

```diff
@@ -0,0 +1,276 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define CATCH_CONFIG_MAIN
+
+#include <memory>
+#include <string>
+#include <set>
+
+#include "../TestBase.h"
+
+#include "processors/GetFile.h"
+#include "python/ExecutePythonProcessor.h"
+#include "processors/LogAttribute.h"
+#include "processors/PutFile.h"
+#include "utils/file/FileUtils.h"
+
+namespace {
+
+#include <unistd.h>
+#define GetCurrentDir getcwd
+
+std::string GetCurrentWorkingDir(void) {
+  char buff[FILENAME_MAX];
+  GetCurrentDir(buff, FILENAME_MAX);
+  std::string current_working_dir(buff);
+  return current_working_dir;
+}
+
+class ExecutePythonProcessorTestBase {
+ public:
+    ExecutePythonProcessorTestBase() :
+      logTestController_(LogTestController::getInstance()),
+      logger_(logging::LoggerFactory<org::apache::nifi::minifi::python::processors::ExecutePythonProcessor>::getLogger()) {
+      reInitialize();
+    }
+    virtual ~ExecutePythonProcessorTestBase() {
+      logTestController_.reset();
+      logTestController_.setDebug<TestPlan>();
+      logTestController_.setDebug<minifi::python::processors::ExecutePythonProcessor>();
+      logTestController_.setDebug<minifi::processors::PutFile>();
+      logTestController_.setDebug<minifi::processors::PutFile::ReadCallback>();
+    }
+
+ protected:
+    void reInitialize() {
+      testController_.reset(new TestController());
+      plan_ = testController_->createPlan();
+    }
+
+    std::string createTempDir() {
+      char dirtemplate[] = "/tmp/gt.XXXXXX";
+      std::string temp_dir = testController_->createTempDirectory(dirtemplate);
+      REQUIRE(!temp_dir.empty());
+      struct stat buffer;
+      REQUIRE(-1 != stat(temp_dir.c_str(), &buffer));
+      REQUIRE(S_ISDIR(buffer.st_mode));
+      return temp_dir;
+    }
+
+    std::string putFileToDir(const std::string& dir_path, const std::string& file_name, const std::string& content) {
+      std::string file_path(dir_path + utils::file::FileUtils::get_separator() + file_name);
+      std::ofstream out_file(file_path);
+      if (out_file.is_open()) {
+        out_file << content;
+        out_file.close();
+      }
+      return file_path;
+    }
+
+    std::string createTempDirWithFile(const std::string& file_name, const std::string& content) {
+      std::string temp_dir = createTempDir();
+      putFileToDir(temp_dir, file_name, content);
+      return temp_dir;
+    }
+
+    std::string getFileContent(const std::string& file_name) {
+      std::ifstream file_handle(file_name);
+      REQUIRE(file_handle.is_open());
+      const std::string file_content{ (std::istreambuf_iterator<char>(file_handle)), (std::istreambuf_iterator<char>())};
+      file_handle.close();
+      return file_content;
+    }
```

Comment:
These are generic enough that they could go to `FileUtils` IMO.

I'd remove the `std::ifstream::close` calls because they are both right before `return`, after which the file is closed anyway. They don't hurt but they're not needed either.

If we handle errors in some way, we shouldn't ignore them IMO. I mean if we check for a good file stream, we should throw if it's not good.

Reply:
1) Sure, but I would prefer to have them in some kind of `TestUtils.h` due to depending on `REQUIRE` macro. Do you want me to create such a file? I can also add the same definition to `FileUtils.h` with a throwing version if that is what you ask in 3).

2) Those will be removed.

3) I am not sure if I understand, does reply 1) apply here?

Reply:
1. I'd replace REQUIRE with check + throw, so that they're reusable and the control flow is the same (tests fail on uncaught exceptions).

3. The first function only operates on the file it the open was successful. If not, there is no indication of the error to the caller. Since there is already some kind of check, I recommend throwing an exception in the else branch.

Reply:
1. I would argue that the workflow would not be any more reusable, as a having a `TestController` is a dependency either way. Also, I would rather have the REQUIRE assertions inside the test-helper function for two reasons:
 - Someone can easily forget adding EXPECT_THROW on the call side ending up accidentaly satisfying "EXPECT_THROW" calls on an upper call stack.
 - Adding the expectation for throwing would make the code would look like we test the helper functions inside of unit tests, and might not be obvious why is there an expectation for not throwing for a given test.